Qt checks if glibc is available and if not disables large part of Qt5WebEngine and thus cripples functionality.
However these parts work fine with Musl so there is no need to disable them.
Just remove the check so it builds  again. Since 5.15.1 is the last version of Qt to be released with this build system it'll be obsolete with Qt6 and there is no real point in fixing this upstream.

diff --git a/src/buildtools/configure.json b/src/buildtools/configure.json
index 1ca6214a..3783c223 100644
--- a/src/buildtools/configure.json
+++ b/src/buildtools/configure.json
@@ -375,8 +375,7 @@
                          && features.webengine-python2
                          && (!config.sanitizer || features.webengine-sanitizer)
                          && (!config.linux || features.pkg-config)
-                         && (!config.linux || features.webengine-host-pkg-config)
-                         && (!config.linux || features.webengine-system-glibc)
+                         && (!config.linux || features.webengine-host-pkg-config) 
                          && (!config.linux || features.webengine-system-khr)
                          && (!config.linux || features.webengine-system-nss)
                          && (!config.linux || features.webengine-system-dbus)

